src/main/java/dev/joopie/jambot/api/spotify/ApiSpotifyService.java

```diff
@@ -69,46 +77,35 @@ public Optional<String> getTrack(String link) {
         return Optional.empty();
     }
 
-    public List<String> getTracksFromPlaylist(String link) {
+    public Optional<Track> searchForTrack(String artistName, String trackName) {
         if (spotifyApi == null || spotifyApi.getAccessToken().isEmpty() || isAccessTokenExpired()) {
             initSpotifyAccessToken();
         }
-        final var playlistId = getSpotifyIdFromLink(link);
-
-        if (playlistId.isEmpty()) {
-            return List.of();
-        }
-        final var playlistsItemsRequest = spotifyApi.getPlaylistsItems(playlistId.get()).build();
-
+        final String searchQuery = String.format("%s - %s", artistName, trackName);
         try {
-            final var playlist = playlistsItemsRequest.execute();
-            return Arrays.stream(playlist.getItems())
-                    .map(playlistItem -> (Track) playlistItem.getTrack())
-                    .map(this::getFormattedTrack)
-                    .toList();
+            List<se.michaelthelin.spotify.model_objects.specification.Track> searchResult =
+                    Arrays.stream(spotifyApi.searchTracks(searchQuery).build().execute().getItems())
+                            .filter(track -> Arrays.stream(track.getArtists())
+                                    .anyMatch(artist -> artist.getName().toLowerCase().contains(artistName.toLowerCase())))
+                            .toList();
+
+            return searchResult.isEmpty() ? Optional.empty() : Optional.of(spotifyAPIConverterService.saveAPIResult(searchResult.getFirst()));
 
         } catch (IOException | SpotifyWebApiException | ParseException e) {
-            log.error(e.getMessage(), e);
-            return List.of();
+            log.error("Error while fetching the Spotify API", e);
+            return Optional.empty();
         }
     }
 
-    private String getFormattedTrack(Track track) {
-        var artists = Arrays.stream(track.getArtists())
-                .map(ArtistSimplified::getName)
-                .collect(Collectors.joining(","));
-        return "%s-%s".formatted(artists, track.getName());
-    }
 
-    private Optional<String> getSpotifyIdFromLink(String link) {
-        final var matcher = SPOTIFY_URL_PATTERN.matcher(link);
 
-        if (matcher.find()) {
-            return Optional.of(matcher.group(2));
-        } else {
-            log.debug("No ID found in the link.");
-            return Optional.empty();
-        }
+    private Optional<String> getSpotifyIdFromLink(String link) {
+        return Optional.of(SPOTIFY_URL_PATTERN.matcher(link))
```

Comment:
In dit geval had ik het met een if-statement gedaan.

Matcher returned altijd een waarde.
De find/matches methode geven aan of iets is gevonden.